packages/react-reconciler/src/ReactFiberUnwindWork.js

```diff
@@ -215,12 +217,19 @@ function throwException(
               : renderExpirationTime;
 
           // Attach a listener to the promise to "ping" the root and retry.
-          const onResolveOrReject = retrySuspendedRoot.bind(
+          let onResolveOrReject = retrySuspendedRoot.bind(
             null,
             root,
             workInProgress,
             pingTime,
           );
+          if (enableInteractionTracking) {
+            // Wrap suspense callback so that the current interaction are preserved.
+            // Interaction threads are unique per root and expiration time.
+            const threadID =
+              renderExpirationTime * 1000 + root.interactionThreadID;
```

Comment:
Will it break if I change the formula here but not in the other place it's used?

Reply:
It will "break" in that the callbacks will receive a different thread ID and so any reporting done based on those callbacks would be inaccurate. I'll add a comment to each of these places noting that we shouldn't change one without changing the others. Good call!